src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs

```diff
@@ -725,8 +718,7 @@ internal void FinishOperationSuccess(SocketError socketError, int bytesTransferr
                     // Mark socket connected.
                     if (socketError == SocketError.Success)
                     {
-                        if (s_loggingEnabled)
-                            SocketsEventSource.Connected(_currentSocket, _currentSocket.LocalEndPoint, _currentSocket.RemoteEndPoint);
+                        NetEventSource.Connected(_currentSocket, _currentSocket.LocalEndPoint, _currentSocket.RemoteEndPoint);
```

Comment:
> We could say that every single call site to any tracing method in NetEventSource needs to have an IsEnabled check before it. It'll lead to clunkier code, but it'll be easy to explain. Is that what you'd prefer?

I think I'm leaning that way, yeah.

I'm used to using macros to do this in native code, which is pretty easy to get right, and not very crufty.  Unfortunately this isn't possible in managed code.

The problem is the "non-trivial work happening" part.  This seems fairly easy to get wrong.  The "no value types" rule in particular made me pause a few times when reviewing the code, since you need to think about the types of the arguments you're passing.

I realize that the value types mistake will get caught at your checks, but it seems like it would be better to have the code be correct at a glance rather than having to think about it.

And other potential issues, like "NetEventSource.Info(this, SomeExpensiveFunction())" can't be caught by your checks.

I realize that that makes the code cruftier, and I don't like that, but overall I think I prefer the more verbose code.